app/models/piece.rb

```diff
@@ -1,3 +1,102 @@
 class Piece < ApplicationRecord
   belongs_to :game
+  validates :color, presence: true
+  validates :type, presence: true
+
+  scope :white, ->() { where(color: 'white') }
+  scope :black, ->() { where(color: 'black') }
+
+  def self.at(x_position, y_position)
+    Piece.find_by(x_position: x_position, y_position: y_position)
+  end
+
+  def exist?(x, y)
+    Piece.where(x_position: x, y_position: y).present?
+  end
+
+  def check_path(x1, y1, x2, y2)
+    if y1 == y2
+      'horizontal'
+    elsif x1 == x2
+      'vertical'
+    else
+      # move diagonal
+      @slope = (y2 - y1).to_f / (x2 - x1).to_f
```

Comment:
See if you can refactor this so that you're not using an instance variable within a model.

Instance variables "hold state" and can make reading code difficult because your methods depend on their value being set before the method is called (forcing methods to be called in a specific order)

😄